### ISSUE TITLE
Does MapView only draw PathViews that are visible?

### ISSUE THREAD

**Issue by mohlendo** [moagrius/MapView#43]
In my use-case I have a lot of stuff that should be drawn on top of the map. Does the MapView only draw those PathViews that are visible on the screen?

**Comment by moagrius** [comment]
No, only tiles are managed relative to the viewport, everything else is rendered normally.  Are you concerned about memory?

**Comment by mohlendo** [comment]
Yes, when I use the ParthViews the MapView does not respond anymore because I have A LOT of PathViews. I have now added a way to draw directly on the bitmap of the tiles which is much faster. This will eventually result in another pull request ;-)

**Comment by moagrius** [comment]
sounds cool - looking forward to seeing it

**Comment by moagrius** [comment]
How'd this go @mohlendo?  In another dev's project, they found that adding/removing paths dynamically was buggy, but I found redrawing was stable, so for the next major update I'm going to rewrite the whole path drawing strategy #57  If you've found anything useful, LMK.  Thanks.

**Comment by phil-applegate** [comment]
Just to add a comment to this, I've recently been trying to draw a particularly large path. When trying to use hardware acceleration to draw the MapView I am presented with the error `OpenGLRenderer: "Shape path too large to be rendered into a texture"` and the path is not drawn.

This can be resolved by disabling hardware acceleration at a cost of performance. Not sure if your update to the path drawing strategy in #57 is planned to solve this but could be worth looking at.

Cheers and thanks.

**Comment by moagrius** [comment]
@phil-applegate ideally i'd like to limit the shape to the viewport, but i'm not sure how realistic that'll end up being.  will definitely keep the note above handy tho, thanks for the input.

**Comment by phil-applegate** [comment]
I've been doing some more research into this issue and found this [post](http://stackoverflow.com/a/15208783/1344221) which looks to be similar to the problem I described above. 

I've modified my local version of `MapView` to use `drawLines()` combined with `drawCircle()` rather than `drawPath()`. As a result the Path does not look quite as good as it did with `drawPath()` but it can now be hardware accelerated thus improving performance and smoothness. 

Hope the link proves to be some use!
